okhttp/src/main/java/okhttp3/internal/http2/Http2Connection.java

```diff
@@ -725,6 +725,7 @@ private void applyAndAckSettings(final Settings peerSettings) {
       synchronized (Http2Connection.this) {
         streamsCopy = streams.values().toArray(new Http2Stream[streams.size()]);
         shutdown = true;
+        Http2Connection.this.lastGoodStreamId = lastGoodStreamId;
```

Comment:
I think this is the line that’s most problematic. The local field `lastGoodStreamId` refers to remotely-initiated streams, and `goAway` is telling us an ID for locally-initiated streams